### PR TITLE
Interim Kubeflow Steering Committee proposal

### PR DESCRIPTION
**Pull request by james-jwu**
Includes skeleton docs for the 3 committees.

### CODE REVIEW

**Review by theadactyl — APPROVED** [review]
lgtm

**Comment by akgraner** [comment]
I have questions/concerns about how this will work? 

How will the "Approved/appointed" Committees will be selected?  Will the interim committee be able to appoint/approve themselves to the "Approved/appointed" Committees? 

I assume the approved/appointed committees will only be in place until the first community election? Will the charter for each of these committees include how each moves from "interim" to an "approved/appointed" to an "elected" committee?   

Will the approved/appointed committee go into effect once we are officially in incubation? Or when we graduate?  

Will all the above be part of the charters and what are the thoughts about the interim - approved/appointed - election processes and timelines?

**Comment by johnugeorge** [comment]
> I have questions/concerns about how this will work?
> 
> How will the "Approved/appointed" Committees will be selected? Will the interim committee be able to appoint/approve themselves to the "Approved/appointed" Committees?
> 
> I assume the approved/appointed committees will only be in place until the first community election? Will the charter for each of these committees include how each moves from "interim" to an "approved/appointed" to an "elected" committee?
> 
> Will the approved/appointed committee go into effect once we are officially in incubation? Or when we graduate?
> 
> Will all the above be part of the charters and what are the thoughts about the interim - approved/appointed - election processes and timelines?

+1  Can we add more details regarding the process/timelines for better clarity?

**Comment by theadactyl** [comment]
Thanks for your questions! I've outlined more details re: roles & timeline below. 

The Interim Steering Committee (IKSC) members are appointed by Google. The Interim Steering Committee is responsible for stewarding the CNCF donation process and creating the initial charters for the steering committees (KSC, KCC, KTOC), including the decision processes, committee roles/composition, and the election process. 

We plan to leverage the existing Knative committee charters where it makes sense. As details are fleshed out, we will update these governance documents. Some areas of the charter will be left for the KSC to define post-transition. 

If/once Kubeflow is accepted into the CNCF, the IKSC will *then* conduct elections for KSC, KCC, and KTOC membership. After elections, further changes to the KSC charter can be made through the established decision process. 

- **Q4 2022** Interim Kubeflow Steering Committee established
- **Q4 2022 - Q1 2023** Preparation for CNCF submission review (updating community docs, license files, etc to accurately reflect *current project state*). Community leads, IKSC provides guidance. 
- **Q1 2023** IKSC creates initial charters & supporting governance documents for KSC, KTOC, KCC
- **? 2023 (Unclear due to backlog)** CNCF submission review begins. IKSC supports, Google is main Point of Contact with CNCF. 
- **? 2023** CNCF submission resolves
- **? 2023** (Assuming acceptance) IKSC holds elections for KSC, KTOC, KCC

**Comment by akgraner** [comment]
Thanks @theadactyl for the answers, but it still doesn't the question of will the Interim KSC appoint themselves as the KSC for incubation stage.  

You mention that, "If/once Kubeflow is accepted into the CNCF, the IKSC will then conduct elections for KSC, KCC, and KTOC membership," but what does accepted mean? Does it mean "accepted" into incubation or "accepted" mean graduated? (Questions, I've been asked by others as well)
 
While you reference KNative as the example you all (IKSC) will use to establish the charters has this approach been shared publicly with links to those charters with the community to also weigh in on? If not, I'd like to suggest we share the KNative charters with the community and get feedback before implementing the above. Not trying to hold up the process, but wanting to inform the community on exactly what that means and be as transparent as possible. 

Super excited for this to happen, but think we need more public feedback from the community. :-)

**Comment by james-jwu** [comment]
Hi Amber, **accepted** means passing the CNCF due diligence review and being accepted into **incubation stage**.

Here is the overall governance document of Knative, which contains links to the 3 committees: https://github.com/knative/community/blob/main/GOVERNANCE.md. I will also add a reference to the Knative docs in the PR.

To your first question about IKSC membership in the KSC committee: this will be proposed in later PRs to the governance documentation. The IKSC will need to discuss the issue. My current proposal is after acceptance of Kubeflow into incubation, IKSC members and X elected members will form the 1st KSC. All KSC seats will be open for election in the subsequent election after Y months. This is a process adopted by Knative as well, and seems to provide a good model for ensuring continuity and consistency of governance.

**Comment by james-jwu** [comment]
I made a couple of updates to address your feedback. PTAL @akgraner @annajung

File: proposals/STEERING-COMMITTEE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+1. Establish KSC, KCC and KTOC
+1. Create committee charters
+1. Work with CNCF TOC to complete the due diligence review
+
+## Kubeflow Steering Committee
+The governance of Kubeflow is an open, living process and document, and will continue to evolve as the community and project change.
+
+The Interim KSC is chartered to establish KSC. The interim committee will propose details of the KSC charter, roles, processes and limitations under this document.
+
+KSC will replace the interim committee after CNCF accepts Kubeflow as an incubation project.
```
Comment (annajung) [inline proposals/STEERING-COMMITTEE.md:23]: Could we specify that KSC will be formed based on the process defined by the interim KSC? As it reads, it sounds like KSC will be established by the interim KSC, but it's the process and the charter for which the interim KSC is responsible for, not the actual choosing of the members. If that's correct, I think additional clarification would be helpful. 

```suggestion
KSC will be formed based on the process defined by the interim KSC and will replace the interim committee after CNCF accepts Kubeflow as an incubation project.
```

Comment (james-jwu) [reply proposals/STEERING-COMMITTEE.md:23]: Sounds good.

File: proposals/STEERING-COMMITTEE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+*[This document is work in progress. It will be moved to https://github.com/kubeflow/community/STEERING-COMMITTEE.md upon completion and approval.]*
+
+The Kubeflow Steering Committee (KSC) is responsible for the general health of the Kubeflow community.
+
+## Interim KSC
+During the CNCF incubation period, KSC will be a small council that consists of the following people:
```
[the remark this answers is not in the document]
Comment (james-jwu) [reply proposals/STEERING-COMMITTEE.md]: Sounds good!

**Comment by johnugeorge** [comment]
Thanks @james-jwu @theadactyl 

LGTM

**Comment by akgraner** [comment]
Thanks @james-jwu! 

Thanks @theadactyl, @james-jwu and @jbottum, I'm confident you all will take input from the community and together we'll all be able to achieve great things for Kubeflow. 

Looking forward to Kubeflow being formally part of the CNCF!

**Comment by james-jwu** [comment]
Thanks everyone for the feedback!
/lgtm
/approve

**Pull request merged**